ui/components/multichain/asset-picker-amount/asset-picker-amount.tsx

```diff
@@ -135,6 +137,10 @@ export const AssetPickerAmount = ({
     borderColor = BorderColor.primaryDefault;
   }
 
+  const shouldHideMaxClearButton =
```

Comment:
Can we rename this to `isSwapAndSendFromNative` or something similar for clarity?

Reply:
I believe the current name of the flag effectively indicates its purpose, which is to hide the max button only under the following condition. Though you are indicating it's not clear, can you help me understand what specifically is unclear about the current name?

Reply:
I think it's more clear to name boolean variables based on tracked behavior vs the resultant action. Naming this `shouldHideMaxClearButton` doesn't account for a) other conditions hiding the max clear button (i.e. `shouldHideMaxClearButton` could be `false`, but we would hide it anyways if there isn't an amount handler available) and b) makes it less clear that we can use this variable for other implementations in the future.